nano/node/active_transactions.cpp

```diff
@@ -770,37 +770,65 @@ std::unique_ptr<nano::container_info_component> nano::recently_confirmed_cache::
  */
 
 nano::recently_cemented_cache::recently_cemented_cache (std::size_t max_size_a) :
-	max_size{ max_size_a }
+	handle (rsnano::rsn_recently_cemented_cache_create1 (max_size_a))
+{
+}
+
+nano::recently_cemented_cache::recently_cemented_cache (nano::recently_cemented_cache const & other_a) :
+	handle (rsnano::rsn_recently_cemented_cache_clone (other_a.handle))
 {
 }
 
+nano::recently_cemented_cache::~recently_cemented_cache ()
+{
+	if (handle != nullptr)
+		rsnano::rsn_recently_cemented_cache_destroy (handle);
+}
+
+nano::recently_cemented_cache & nano::recently_cemented_cache::operator= (const nano::recently_cemented_cache & other_a)
+{
+	if (handle != nullptr)
+		rsnano::rsn_recently_cemented_cache_destroy (handle);
+
+	handle = rsnano::rsn_recently_cemented_cache_clone (other_a.handle);
+	return *this;
+}
+
 void nano::recently_cemented_cache::put (const nano::election_status & status)
 {
-	nano::lock_guard<nano::mutex> guard{ mutex };
-	cemented.push_back (status);
-	if (cemented.size () > max_size)
-	{
-		cemented.pop_front ();
-	}
+	rsnano::rsn_recently_cemented_cache_put (handle, status.handle);
 }
 
 nano::recently_cemented_cache::queue_t nano::recently_cemented_cache::list () const
 {
-	nano::lock_guard<nano::mutex> guard{ mutex };
-	return cemented;
+	rsnano::RecentlyCementedCachedDto recently_cemented_cache_dto;
+	rsnano::rsn_recently_cemented_cache_list (handle, &recently_cemented_cache_dto);
+	nano::recently_cemented_cache::queue_t result;
+	rsnano::ElectionStatusHandle * const * current;
+	int i;
+	for (i = 0, current = recently_cemented_cache_dto.items; i < recently_cemented_cache_dto.count; ++i)
+	{
+		nano::election_status election_status (reinterpret_cast<const rsnano::ElectionStatusHandle *> (current));
```

Comment:
I dont think this is correct. By using a reinterpret_cast you would interpret an `ElectionStatusHandle**` as `ElectionStatusHandle*`. This is not tested, but I think this would work: 
```nano::election_status election_status (*current);```